docker/publish.sh

```diff
@@ -4,9 +4,9 @@ set -euo pipefail
 # Cd into directory holding this script
 cd "${BASH_SOURCE[0]%/*}"
 
-image=${1:-"flexflow-cuda"}
-if [[ "${image}" != @(flexflow-environment-cuda|flexflow-environment-hip_cuda|flexflow-environment-hip_rocm|flexflow-environment-intel|flexflow-cuda|flexflow-hip_cuda|flexflow-hip_rocm|flexflow-intel) ]]; then
-  echo "Error, image name ${image} is invalid. Choose between 'flexflow-environment-{cuda,hip_cuda,hip_rocm,intel}' and 'flexflow-{cuda,hip_cuda,hip_rocm,intel}'."
+image=${1:-"specinfer-cuda"}
```

Comment:
Here, this script will be used by all branches, so we should make sure to still accept image names with the `flexflow-` prefix. In addition, we will need to add one important modification. For the normal branches, if the argument is `flexflow-{cuda,hip_cuda,hip_rocm,intel}`, you will make sure that the image exists in the system and publish it. If the argument is `specinfer-{cuda,hip_cuda,hip_rocm,intel}`, on the other hand, an image with that name won't exist in the system. We could also change the `build.sh` script to build images named `specinfer` for the inference branch, but it's more laborious, so I think an easier way to do this is to simply rename the images from `flexflow-{cuda,hip_cuda,hip_rocm,intel}` to `specinfer-{cuda,hip_cuda,hip_rocm,intel}` when we are in the inference branch. Hint: you can do this by modifying the instruction: `docker tag "$image":latest ghcr.io/flexflow/"$image":latest` below. 

Finally, since we are not publishing the environment images for the inference branch, you may want to add a check and print an error if the user (or a workflow) attempts to publish an image named `specinfer-environment-{cuda,hip_cuda,hip_rocm,intel}`

Reply:
I understand that in inference branch, images named specinfer-{cuda,hip_cuda,hip_rocm,intel} won't exist. Therefore, do you mean we can chnage the "docker tag "$image":latest ghcr.io/flexflow/"$image":latest" to "if in inference branch, then: docker tag "$image":latest ghcr.io/flexflow/specinfer-{cuda,hip_cuda,hip_rocm,intel}:latest" no matter what input name it is?